v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/BigQueryDynamicDestinations.java

```diff
@@ -68,35 +78,34 @@ private BigQueryDynamicDestinations(
     this.useStorageWriteApi = bigQueryDynamicDestinationsOptions.getUseStorageWriteApi();
   }
 
-  private TableId getTableId(String bigQueryTableTemplate, TableRow tableRow) {
+  private String getTableName(TableRow tableRow) {
     String bigQueryTableName =
         BigQueryConverters.formatStringTemplate(bigQueryTableTemplate, tableRow);
 
-    return TableId.of(bigQueryProject, bigQueryDataset, bigQueryTableName);
+    return String.format("%s:%s.%s", bigQueryProject, bigQueryDataset, bigQueryTableName);
   }
 
   @Override
-  public KV<TableId, TableRow> getDestination(ValueInSingleWindow<TableRow> element) {
-    TableRow tableRow = element.getValue();
-    return KV.of(getTableId(bigQueryTableTemplate, tableRow), tableRow);
+  public Coder<KV<String, List<TableFieldSchema>>> getDestinationCoder() {
```

Comment:
May we trouble you to add a test for whether the `Coder<KV<String, List<TableFieldSchema>>>` decode/encode yeild matching results, particularly in the setting of a `RECORD` `getType()` field where said RECORD has itself multiple fields?

Reply:
The `codecov/patch` failure among the checks reminded me of this.

Reply:
Added unit test for getDestinationCoder